src/node/internal/internal_http_client.ts

```diff
@@ -360,15 +354,15 @@ export class ClientRequest extends OutgoingMessage implements _ClientRequest {
     }
 
     const host = this.getHeader('host') ?? this.host;
-    const url = new URL(`http://${host}`);
+    let url = new URL(`http://${host}`);
     url.protocol = this.protocol;
     url.port = this.port;
-    url.pathname = this.path;
-    if (this.#options.search != null) {
-      url.search = this.#options.search;
-    }
-    if (this.#options.hash != null) {
-      url.hash = this.#options.hash;
```

Comment:
It would also be worth removing these properties from the `urlToHttpOptions` helper as well if they're not defined in these APIs at all.

Reply:
Ok thanks for clarifying, makes sense if they are part of the public API.